go/login/server/server.go

```diff
@@ -0,0 +1,299 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package backend implements GLOME-login server framework.
+package server
+
+import (
+	"encoding/base64"
+	"fmt"
+	"log"
+	"net/http"
+	"net/url"
+	"strconv"
+	"sync"
+
+	"github.com/google/glome/go/glome"
+	"github.com/gorilla/mux"
+)
+
+const (
+	// MaxResponseSize is the maximum size in characaters of the response token
+	MaxResponseSize = 44
+)
+
+var (
+	// ErrInvalidResponseLen  denotes that a ephemeral is invalid.
+	ErrInvalidResponseLen = fmt.Errorf("invalid response length provided")
+
+	// This error are hypothesis on what will be provided by the glome login lib
+
+	// ErrFailedHandshake denotes that GLOME handshake has failed.
+	ErrFailedHandshake = fmt.Errorf("400 bad request: failed handshake")
+	// ErrVersionNotSupported denotes that a version of GLOME handshake is not supported.
+	ErrVersionNotSupported = fmt.Errorf("400 bad request: version not supported")
+	// ErrInvalidPrefixType denotes that a prefix type is invalid.
+	ErrInvalidPrefixType = fmt.Errorf("400 bad request: invalid prefix type")
+	// ErrInvalidPrefix7 denotes that a prefix7 is invalid.
+	ErrInvalidPrefix7 = fmt.Errorf("404 Not Found: prefix7 not found")
+	// ErrInvalidEph denotes that a ephemeral is invalid.
+	ErrInvalidEph = fmt.Errorf("400 bad request: invalid ephemeral Key")
+	// ErrIncorrectTag denotes that received tag is incorrect
+	ErrIncorrectTag = fmt.Errorf("invalid Tag")
```

Comment:
Shouldn't all of these come from glome-login instead?

Also generally when errors are returned to the user you use `http.Error()` with `http.StatusBadRequest` to actually send the error code back, rather than mentioning it just in the text. All of these look like invalid user input so there is no need to distinguish between them and you can treat all of them equally.

Reply:
Yes! They were using during development only, are to be deprecated  as soon as integration with go-glome-login is done. On `http.Error()`: will take note and modify asap.